lib/evm/gas.ex

```diff
@@ -123,16 +123,20 @@ defmodule EVM.Gas do
 
       iex> EVM.Gas.instr_cost(:extcodesize, nil, nil, nil)
       700
+
   """
   @spec instr_cost(atom(), EVM.state, MachineState.t, ExecEnv.t) :: t | nil
   def instr_cost(:sstore, state, machine_state, _exec_env), do: cost_sstore(state, machine_state)
   def instr_cost(:exp, _state, machine_state, _exec_env) do
     case Enum.at(machine_state.stack, 1) do
       0 -> @g_exp
-      s -> @g_exp + @g_expbyte * ( 1 + MathHelper.floor( MathHelper.log(s, 256) ) )
+      s -> @g_exp + @g_expbyte * byte_size(:binary.encode_unsigned(s))
```

Comment:
Hmm, trying to wrap my head about this, from: 
<img width="363" alt="screen shot 2017-08-27 at 5 23 08 pm" src="https://user-images.githubusercontent.com/271024/29755412-6dc38a74-8b4c-11e7-9552-8b6a3dd35c5d.png">

Reply:
So `( 1 + MathHelper.floor( MathHelper.log(s, 256) )` == `ceil(log256(s))`

`ceil(log256(1)) == 1`
`ceil(log256(256)) == 2`

also
`byte_size(1) == 1`
`byte_size(256) == 2`

I ran through a bunch of ints to make sure:

    Enum.reduce(1..1024, true, fn(n, acc) ->
      acc && byte_size(:binary.encode_unsigned(n)) == ( 1 + MathHelper.floor( MathHelper.log(n, 256) ) )
    end)

Using `byte_size` seemed more descriptive to me. Each byte costs 3 gas. I agree this is harder less descriptive when comparing to the yellow paper though. I'll change it back and we can discuss in a follow on PR if we want to.

Reply:
Ooops, oh yeah what we have currently fails the `exp0` test.

Going to leave it as `byte_size` since that passes the tests and we can change it back in a follow on PR if we want to more closely match the yellow paper.